gtda/utils/validation.py

```diff
@@ -10,22 +10,21 @@
     'wasserstein': [('p', int, (1, np.inf)),
                     ('delta', numbers.Number, (1e-16, 1.))],
     'betti': [('p', numbers.Number, (1, np.inf)),
-              ('n_values', int, (1, np.inf))],
+              ('n_bins', int, (1, np.inf))],
     'landscape': [('p', numbers.Number, (1, np.inf)),
-                  ('n_values', int, (1, np.inf)),
+                  ('n_bins', int, (1, np.inf)),
                   ('n_layers', int, (1, np.inf))],
-    'heat': [('order', numbers.Number, (1, np.inf)),
-             ('n_values', int, (1, np.inf)),
+    'heat': [('p', numbers.Number, (1, np.inf)),
+             ('n_bins', int, (1, np.inf)),
              ('sigma', numbers.Number, (0., np.inf))],
-    'persistent_image': [('order', numbers.Number, (1, np.inf)),
-                         ('n_values', int, (1, np.inf)),
-                         ('sigma', numbers.Number, (0., np.inf)),
-                         ('weight_function', types.FunctionType,
-                          None)]}
+    'persistence_image': [('p', numbers.Number, (1, np.inf)),
+                          ('n_bins', int, (1, np.inf)),
+                          ('sigma', numbers.Number, (0., np.inf)),
+                          ('weight_function', types.FunctionType,
+                           None)]}
 
-available_metric_params = list(set(
-    [param for param_list in available_metrics.values()
-     for (param, param_type, param_range) in param_list]))
+available_metric_params = {metric: [p[0] for p in param_lst]
+                           for metric, param_lst in available_metrics.items()}
```

Comment:
@gtauzin FYI
I hear that you are working on `validate_params`. I noticed some questionable behavior, and the changes in that method fix that. However, I do not claim that it is an attempt to make it 'nice'.

Reply:
`validate_params` is completely broken. It needs to be fixed, documented, and the consequences of this fix will mean that a number of hyperparameter validations will have to be fixed too throughout the code.

I can take care of the fixing and the commenting as I wrote it, but if you have time, could you help with the debugging that will follow?

I will open a PR.

Reply:
I am happy to work on this, in whatever capacity: from writing the function, to debugging. Of course, with your reviews.
Just let me know.